pkg/arvo/sys/vane/ames.hoon

```diff
@@ -1105,31 +1217,62 @@
     ^+  event-core
     ::  relay the vane ack to the foreign peer
     ::
-    =+  ^-  [her=ship =bone]  (parse-bone-wire wire)
-    ::
+    ?~  parsed=(parse-bone-wire wire)
+      ::  no-op
+      ::
+      =/  =tape  "ames dropping malformed wire: {(spud wire)}"
+      (emit duct %pass /malformed-wire %d %flog %text tape)
+    ?>  ?=([@ her=ship *] u.parsed)
+    =*  her          her.u.parsed
     =/  =peer-state  (got-peer-state her)
     =/  =channel     [[our her] now channel-state -.peer-state]
     =/  peer-core    (make-peer-core peer-state channel)
+    |^
+    ?:  ?&  ?=([%new *] u.parsed)
+            (lth rift.u.parsed rift.peer-state)
+        ==
+      ::  ignore events from an old rift
+      ::
+      ?.  odd.veb  event-core
+      (log "ames dropping old rift wire: {(spud wire)}")
+    =/  =bone
+      ?-(u.parsed [%new *] bone.u.parsed, [%old *] bone.u.parsed)
+    =?  event-core.peer-core  &(odd.veb ?=([%old *] u.parsed))
+      (log "ames parsing old wire format: {(spud wire)}")
+    ?~  error
+      (send-ack bone)
+    (send-nack bone u.error)
+    ::
+    ++  log
+      |=  =tape
+      ^+  event-core
+      (emit duct %pass /on-take-done-parse-wire %d %flog %text tape)
```

Comment:
I think it's better to use `+trace`, because then it'll print even if the event crashes, and it'll print in order with other things from ames-verb or from `|verb`

Reply:
Sounds good—I mostly added it so I could then capture that in the tests, but I guess I could look instead at how many events are emitted